plugins/utilities/natpmp_upnp.py

```diff
@@ -0,0 +1,256 @@
+            shutil.copytree(upnpy_source_dir, upnpy_file_path)
+            shutil.rmtree(Path(f"{install_path}/NAT-PMP-1.3.2"))
+            shutil.rmtree(Path(f"{install_path}/UPnPy-1.1.8"))
+        except Exception as e:
+            if type(e) == shutil.Error:
+                shutil.rmtree(Path(f"{install_path}/NAT-PMP-1.3.2"))
+                shutil.rmtree(Path(f"{install_path}/UPnPy-1.1.8"))
+            else:
+                pass
+
+        # Patch to natpmp to work without netifaces
+        with open(f"{nat_pmp_file_path}/__init__.py", "r") as f:
+            lines = f.readlines()
+            # Define the new function as a string
+            new_function = '''
+# Plucked from https://github.com/tenable/upnp_info/blob/d20a1fda8ca4877d61b89fe7126077a3a5f0b322/upnp_info.py#L23
+def get_gateway_addr():
+    """Returns the gateway ip of the router if upnp service is available"""
+    locations = set()
+    location_regex = re.compile("location:[ ]*(.+)"+ chr(13) + chr(10), re.IGNORECASE)
+    ssdpDiscover = (
+        "M-SEARCH * HTTP/1.1"+ chr(13) + chr(10)
+        + "HOST: 239.255.255.250:1900"+ chr(13) + chr(10)
+        + 'MAN: "ssdp:discover"'+ chr(13) + chr(10)
+        + "MX: 1"+ chr(13) + chr(10)
+        + "ST: ssdp:all"+ chr(13) + chr(10)
+        + chr(13) + chr(10)
+    )
+
+    sock = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+    sock.sendto(ssdpDiscover.encode("ASCII"), ("239.255.255.250", 1900))
+    sock.settimeout(3)
+    try:
+        while True:
+            data, addr = sock.recvfrom(1024)  # buffer size is 1024 bytes
+            location_result = location_regex.search(data.decode("ASCII"))
+            if location_result and (location_result.group(1) in locations) == False:
+                locations.add(location_result.group(1))
+    except socket.error:
+        sock.close()
+    if locations:
+        for location in locations:
+            parsed_url = urlparse(location)
+            if parsed_url.path.endswith("xml"):
+                gateway_ip_address = parsed_url.netloc.split(':')[0]
+                return gateway_ip_address
+                 
+'''
+
+            # Replace the function
+            lines[224:229] = new_function
+            lines[21] = "import socket\nimport re\nfrom urllib.parse import urlparse"
+
+        with open(f"{nat_pmp_file_path}/__init__.py", "w") as f:
+            f.writelines(lines)
+
+    add_port_mapping()
+
+
+@threaded
+def confirm_port():
+    time.sleep(5)
+    with urlopen("https://legacy.ballistica.net/bsAccessCheck") as resp:
+        resp = resp.read().decode()
+        resp = ast.literal_eval(resp)
+        return resp["accessible"]
+
+
+@threaded
+def add_port_mapping():
+    # Try to add UDP port using NAT-PMP
+    import socket
+    import natpmp
+    from natpmp import NATPMPUnsupportedError, NATPMPNetworkError
+
+    try:
+        natpmp.map_port(
+            natpmp.NATPMP_PROTOCOL_UDP,
+            43210,
+            43210,

```

Comment:
do you think we should use `bascenev1.get_game_port()`? since game binds on a diff port if 43210 is taken.